platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/NetworkPeerIdentifier.java

```diff
@@ -60,6 +61,11 @@ public NetworkPeerIdentifier(@NonNull final PlatformContext platformContext, @No
         noPeerFoundLogger = new RateLimitedLogger(logger, platformContext.getTime(), Duration.ofMinutes(5));
         this.x501PrincipalsAndPeers = HashMap.newHashMap(peers.size());
         for (final PeerInfo peerInfo : peers) {
+            if (!CryptoStatic.checkCertificate(peerInfo.signingCertificate())) {
```

Comment:
if a peer does not have a cert, why even add him to the list?

Reply:
Looking closer:  `fetchGossipCaCertificate` does a `CryptoStatic.decodeCertificate` but not a `CryptoStatic.checkCertificate`.    The `List<PeerInfo>` was constructed with non-null certificates that successfully decode from an encoded form.   

If someone modifies the construction of `List<PeerInfo>`, this check would remain.  It's the latest check that enforces the invariants in the issue.